realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -99,6 +99,11 @@
     final ConcurrentHashMap<WeakReference<RealmObjectProxy>, Object> realmObjects =
             new ConcurrentHashMap<WeakReference<RealmObjectProxy>, Object>();
 
+    // List of onSuccess callbacks from async transactions. We need to track all callbacks as notifying listeners might
+    // be delayed due to the presence of async queries. This can mean that multiple async transactions can complete
+    // before we are ready to notify all of them.
+    private final List<Runnable> pendingOnSuccessAsyncTransactionCallbacks = new ArrayList<Runnable>();
```

Comment:
what happen when the user cancel the transaction using the provided `RealmAsyncTask`?
Also is there a risk of memory leak?